calico/src/main/scala/calico/html/NodesInterpolator.scala

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2022 Arman Bilge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package calico.html
+
+import cats.effect.IO
+import cats.effect.Resource
+import cats.syntax.functor.*
+import fs2.concurrent.Signal
+import fs2.dom.HtmlElement
+
+import scala.annotation.nowarn
+
+// Type-level function for interspersing strings between tuple elements
+type IntersperseStrings[T <: Tuple] <: Tuple =
+  T match {
+    case EmptyTuple => String *: EmptyTuple
+    case (t *: ts) => String *: t *: IntersperseStrings[ts]
+  }
+
+// Value-level function for interspersing strings between tuple elements
+inline def intersperseStrings[T <: Tuple](t: T, strings: Seq[String]): IntersperseStrings[T] =
+  inline scala.compiletime.erasedValue[T] match {
+    case _: EmptyTuple => strings.head *: EmptyTuple
+    case _: (head *: tail) =>
+      inline t match {
+        case v: (`head` *: `tail`) =>
+          val (h *: t) = v
+          strings.head *: h *: intersperseStrings(t, strings.tail)
+      }
+  }
+
+// Add this helper method
+private def isSignal(obj: Any): Boolean =
+  obj != null && obj.getClass.getName.contains("fs2.concurrent.Signal")
+
+extension (sc: StringContext) {
+  // String interpolator with variadic arguments (standard string interpolator style)
+  @nowarn("msg=pattern selector should be an instance of Matchable")
+  def nodes(args: Any*): List[NodeContent] = {
```

Comment:
Instead of exposing a `List[NodeContent]`, maybe we hide this implementation detail with a wrapper class `NodeInterpolator`